v1/messages/src/requestHandlers.ts

```diff
@@ -96,11 +107,47 @@ function parseSlugFromURI(uri: string, resource: string): string {
   return "";
 }
 
+function parseAuthorizationHeader(auth: string): string | null {
+  if (auth === undefined) {
```

Comment:
This is a bug because on line 19 of `index.ts`, you use null coalescing so the auth parameter is never `undefined`, only `string | null`. Ultimately this problem is caused by the fact that you are splitting the parsing logic into two places.

Instead, we should put all parse authorization header logic in this function, so I would recommend changing the method signature of this function to take in the request.headers object as a parameter, and then encapsulating the entire auth variable inside this method, since we don't need that auth variable after we extract the authorization id from it.